vllm/v1/worker/gpu_model_runner.py

```diff
@@ -1023,20 +1023,6 @@ def execute_model(
                                                     target_probs,
                                                     sampling_metadata)
 
-        # TODO(woosuk): The following loop can be slow since it iterates over
```

Comment:
@WoosukKwon - I think this logic not needed / incorrect as is

We don't sample a partial prefill so I do not think re-winding is needed

Reply:
Could you elaborate more? We do sample new tokens for all requests in the batch, for simplicity. For partial prefills, we simply discard the generated tokens. However, since this changes the random generator of the request, we take care of it here.

Reply:
You're right, my bad

We can avoid this this loop though since there will only be one partial prefill at a time

Reply:
That's not guaranteed actually. There exist some cases when the batch includes multiple partial prefills.

Reply:
I guess #14905 is the best we can do rn?
I think we can further optimize by skipping the similar logic in scheduler's `update_from_output`.